site_scons/buildutils.py

```diff
@@ -617,92 +704,95 @@ def listify(value):
         # assume `value` is a string
         return value.split()
 
-def removeFile(name):
-    """ Remove file (if it exists) and print a log message """
-    if os.path.exists(name):
-        print('Removing file "%s"' % name)
-        os.remove(name)
 
-def removeDirectory(name):
-    """ Remove directory recursively and print a log message """
-    if os.path.exists(name):
-        print('Removing directory "%s"' % name)
-        shutil.rmtree(name)
+def remove_file(name: Union[Path, str]) -> None:
+    """Remove file (if it exists) and print a log message."""
+    path_name = Path(name)
+    if path_name.exists():
+        logger.info(f"Removing file '{name!s}'")
+        path_name.unlink()
+
+
+def remove_directory(name: Union[Path, str]) -> None:
+    """Remove directory recursively and print a log message."""
+    path_name = Path(name)
+    if path_name.exists() and path_name.is_dir():
+        logger.info(f"Removing directory '{name!s}'")
+        shutil.rmtree(path_name)
+
 
 def ipdb():
     """
     Break execution and drop into an IPython debug shell at the point
     where this function is called.
     """
     from IPython.core.debugger import Pdb
-    from IPython.core import ipapi
+    from IPython.core import getipython
 
-    ip = ipapi.get()
+    ip = getipython.get_ipython()
     def_colors = ip.colors
     Pdb(def_colors).set_trace(sys._getframe().f_back)
 
 
-def getSpawn(env):
+def get_spawn(env):
     """
     A replacement for env['SPAWN'] on Windows that can deal with very long
     commands, namely those generated when linking. This is only used when
     compiling with MinGW, as SCons automatically uses a tempfile for the
     MSVC link command.
 
     Pass the return value of this function as the SPAWN keyword argument to
-    the Library target, e.g.:
+    the Library target, for example:
 
-        env.SharedLibrary(..., SPAWN=getSpawn(env))
+        env.SharedLibrary(..., SPAWN=get_spawn(env))
 
     Adapted from http://www.scons.org/wiki/LongCmdLinesOnWin32
     """
 
-    if 'cmd.exe' not in env['SHELL'] or env.subst('$CXX') == 'cl':
-        return env['SPAWN']
-
-    try:
-        useShowWindow = subprocess.STARTF_USESHOWWINDOW
-    except AttributeError:
-        useShowWindow = subprocess._subprocess.STARTF_USESHOWWINDOW
+    if "cmd.exe" not in env["SHELL"] or env.subst("$CXX") == "cl":
+        return env["SPAWN"]
 
-    def ourSpawn(sh, escape, cmd, args, environ):
-        newargs = ' '.join(args[1:])
+    def our_spawn(sh, escape, cmd, args, environ):
```

Comment:
while this is internal, it is complex enough to use some comments

Reply:
What comments do you think would help improve, beyond the top-level docstring?

Reply:
Looking over this again, I believe it's fine. Some of the context doesn't show up on when looking at the PR on the web interface (and trusty emacs isn't too helpful here either - although someone probably has written an extension). 

Regardless, opening the PR in VS code was a revelation ... :wink: